pkg/api/node.go

```diff
@@ -137,11 +137,9 @@ func (m *nodeMetrics) ConvertToTable(ctx context.Context, object runtime.Object,
 	switch t := object.(type) {
 	case *metrics.NodeMetrics:
 		table.ResourceVersion = t.ResourceVersion
-		table.SelfLink = t.SelfLink
```

Comment:
staticcheck:
> table.SelfLink is deprecated: selfLink is a legacy read-only field that is no longer populated by the system. +optional

Reply:
We cannot remove it willy nilly just because it's legacy. We need to be backward compatible, which this linter seems not understand. Deprecated doesn't mean "can be removed at will", it means "it will not be supported further on, so don't create new usage".

I would prefer to keep the old code to avoid breaking old clients that might rely on it.